test/e2e/tests/confirmations/transactions/increase-token-allowance-redesign.spec.ts

```diff
@@ -0,0 +1,231 @@
+/* eslint-disable @typescript-eslint/no-require-imports, @typescript-eslint/no-var-requires */
+import {
+  largeDelayMs,
+  veryLargeDelayMs,
+  WINDOW_TITLES,
+} from '../../../helpers';
+import { Mockttp } from '../../../mock-e2e';
+import { Driver } from '../../../webdriver/driver';
+import { scrollAndConfirmAndAssertConfirm } from '../helpers';
+import { openDAppWithContract, TestSuiteArguments } from './shared';
+
+const {
+  defaultGanacheOptions,
+  defaultGanacheOptionsForType2Transactions,
+  withFixtures,
+} = require('../../../helpers');
+const FixtureBuilder = require('../../../fixture-builder');
+const { SMART_CONTRACTS } = require('../../../seeder/smart-contracts');
+
+describe('Confirmation Redesign ERC20 Increase Allowance', function () {
+  const smartContract = SMART_CONTRACTS.HST;
+
+  describe('Submit an increase allowance transaction @no-mmi', function () {
+    it('Sends a type 0 transaction (Legacy) with a small spending cap', async function () {
+      await withFixtures(
+        {
+          dapp: true,
+          fixtures: new FixtureBuilder()
+            .withPermissionControllerConnectedToTestDapp()
+            .withPreferencesController({
+              preferences: {
+                redesignedConfirmationsEnabled: true,
+                isRedesignedConfirmationsDeveloperEnabled: true,
+              },
+            })
+            .build(),
+          ganacheOptions: defaultGanacheOptions,
+          smartContract,
+          testSpecificMock: mocks,
+          title: this.test?.fullTitle(),
+        },
+        async ({ driver, contractRegistry }: TestSuiteArguments) => {
+          await openDAppWithContract(driver, contractRegistry, smartContract);
+
+          await createERC20IncreaseAllowanceTransaction(driver);
+
+          const NEW_SPENDING_CAP = '3';
+          await editSpendingCap(driver, NEW_SPENDING_CAP);
+
+          await scrollAndConfirmAndAssertConfirm(driver);
+
+          await assertChangedSpendingCap(driver, NEW_SPENDING_CAP);
```

Comment:
The logic appears the same across these four tests except with varying spending caps. What if we move these into a reusable function? e.g. `createAndAssertIncreaseAllowanceSubmission(updateSpendingCap)`